### PR TITLE
Handle the case where the Windows Update Agent is updating itself. 

### PR DESCRIPTION
**Pull request by johnstoj**
My powershell isn't pretty and it's kinda hard to test this change since it's a bit of a race condition, but I think this should fix issue #267.

I'm open to suggestions on doing this a better way.

### CODE REVIEW

File: Boxstarter.WinConfig/Install-WindowsUpdate.ps1 [hunk 1]
```diff
@@ -138,11 +138,43 @@ Install-WindowsUpdate -GetUpdatesFromMS:`$$GetUpdatesFromMS -AcceptEula:`$$Accep
...
+        catch {
+            # Check for WU_E_SELFUPDATE_IN_PROGRESS
+            if($_.Exception.HResult -eq -2145124325) {
+                if ($currentRetry -gt $retries) {
+                    # We can't wait forever...
+                    Out-BoxstarterLog "Windows Update Agent took too long to update itself."
+                    throw
+                }
+
+                Out-BoxstarterLog "Windows Update Agent is self-updating... Waiting."
```
Comment (mwrock) [inline Boxstarter.WinConfig/Install-WindowsUpdate.ps1]: I'd change this to `Write-BoxstarterMessage` so that it prints to the console.

Comment (johnstoj) [inline Boxstarter.WinConfig/Install-WindowsUpdate.ps1]: Done.

**Comment by mwrock** [comment]
Thanks for this!!

**Comment by johnstoj** [comment]
Thank-you for Boxstarter- It makes my life easier. :)

Any idea when you'll cut a new release? No pressure... Just asking.

**Pull request merged**